CrrExistingObjects/metadataUtils.js

```diff
@@ -85,26 +91,33 @@ function _getNullVersion(objMD, bucketName, objectKey, log, cb) {
 }
 
 function getMetadata(params, log, cb) {
+    const { Bucket, Key } = params;
     let versionId = params.VersionId;
+    log.debug('getting object metadata', {
+        method: 'metadataUtils.getMetadata',
+        bucket: Bucket,
+        objectKey: Key,
+        versionId,
+    });
     if (versionId && versionId !== 'null') {
         versionId = versionIdUtils.decode(versionId);
     }
-
     if (versionId instanceof Error) {
         const errMsg = 'Invalid version id specified';
         return cb(errors.InvalidArgument.customizeDescription(errMsg));
     }
-
     const mdParams = {
         versionId,
     };
-    const { Bucket, Key } = params;
     return metadataClient.getBucketAndObjectMD(Bucket, Key, mdParams, log,
         (err, data) => {
             if (err) {
                 return cb(err);
             }
             const objMD = data.obj ? JSON.parse(data.obj) : undefined;
+            if (objMD === undefined) {
```

Comment:
Out of curiosity, do you know in which case objMD can be undefined?

Reply:
The only case I think of from the implementation is, if the key does not exist.
https://github.com/scality/cloudserver/blob/development/8.2/lib/metadata/metadataUtils.js#L62
https://github.com/scality/Arsenal/blob/development/8.1/lib/storage/metadata/mongoclient/MongoClientInterface.js#L262